src/loader/bgw_launcher.c

```diff
@@ -213,23 +218,38 @@ bgw_cluster_launcher_register(void)
 
 /*
  * Register a background worker that calls the main TimescaleDB background
- * worker launcher library (i.e. loader) and uses the scheduler entrypoint
- * function.  The scheduler entrypoint will deal with starting a new worker,
+ * worker launcher library (i.e. loader) and uses either the staging scheduler
+ * entrypoint of the scheduler entrypoint function.
+ *
+ * The staging scheduler is only used when intercepting CREATE
```

Comment:
I'm not a huge fan of two things here: 
1) the name staging scheduler
2) the fact that it exists. Ideally this would happen in the launcher itself. 
As I see it that can happen relatively easily in one of two ways:
a) when the launcher receives a DATABASE_CREATED message, it returns synchronously, then waits on the vxid itself and launches the scheduler (the create database command is the only one that can exist in the virtual transaction, so it will return quickly, but if we're not comfortable with that, we can do a timeout-wait like we do for other things). 
b) we can add the db and the vxid to another data structure and add a timeout to our main loop and then check the vxid during our main loop and take actions accordingly. 
Either way, we avoid some of the complexity and naming issues here and also avoid launching the separate worker to deal with this case, which we've tried quite hard to avoid in other parts of the code.